### ISSUE TITLE
[release/2.1.2xx] Unit test failing: ItPublishesARidSpecificAppSettingSelfContainedToFalse missing  'Microsoft.NETCore.App' version '2.0.9'

### ISSUE THREAD

**Issue by assyadh** [dotnet/cli#9562]
Hi,

I am trying to build the cli and having one unit test failing, referencing  'Microsoft.NETCore.App' version '2.0.9'

## Steps to reproduce
Check out release/2.1.2xx and build it on Amazon Linux:
./build.sh --linux-portable --configuration Release

## Expected  behavior
Unit tests should pass.
Last successful build on my end:  Jun 14, 2018 12:00 AM

## Actual behavior
One unit tests is failing
218 Warning(s)
1 Error(s)

Content of the .trx file:

```
> Failed   Microsoft.DotNet.Cli.Publish.Tests.GivenDotnetPublishPublishesProjects.ItPublishesARidSpecificAppSettingSelfContainedToFalse
>     15>DOTNETTEST : error Message:  [/build/cli/build/test/RunTest.proj]
>           Expected command to pass but it did not.
>          File Name: /build/cli/artifacts/linux-x64/stage2/dotnet
>          Arguments: /build/cli/test/dotnet-publish.Tests/bin/Release/netcoreapp2.0/PublishesSelfContainedFalse/MSBuildTestApp/bin/Release/netcoreapp2.0/linux-x64/publish/MSBuildTestApp.dll
>          WorkingDir:: 
>          Exit Code: 131
>          StdOut:
>          
>          StdErr:
>          It was not possible to find any compatible framework version
>          The specified framework 'Microsoft.NETCore.App', version '2.0.9' was not found.
>            - Check application dependencies and target a framework version installed at:
>                /
>            - Alternatively, install the framework version '2.0.9'.
>          
>          Stack Trace:
>             at FluentAssertions.Execution.XUnit2TestFramework.Throw(String message)
>             at FluentAssertions.Execution.AssertionScope.FailWith(String message, Object[] args)
>             at Microsoft.DotNet.Tools.Test.Utilities.CommandResultAssertions.Pass() in /build/cli/test/Microsoft.DotNet.Tools.Tests.Utilities/Assertions/CommandResultAssertions.cs:line 32
>             at Microsoft.DotNet.Cli.Publish.Tests.GivenDotnetPublishPublishesProjects.ItPublishesARidSpecificAppSettingSelfContainedToFalse() in /build/cli/test/dotnet-publish.Tests/GivenDotnetPublishPublishesProjects.cs:line 176
```

**Comment by livarcocc** [comment]
What framework version do you have available under `/build/cli/artifacts/linux-x64/stage2/dotnet/shared/microsoft.netcore.app`? Also, not sure why it is targeting 2.0.9. Did you change anything in your build? I see from this file that it should be targeting 2.0.7: https://github.com/dotnet/cli/blob/release/2.1.2xx/build/DependencyVersions.props#L5

**Comment by assyadh** [comment]
Hi,

I did not change anything on the build. 

The framework version available under `/build/cli/artifacts/linux-x64/stage2/dotnet/shared/microsoft.netcore.app` is 2.0.7

I tried to manually run the same test case using the built cli, and facing the same error:

```
bash-4.2# pwd
/build/cli/artifacts/linux-x64/stage2
bash-4.2# ./dotnet /build/cli/test/dotnet-publish.Tests/bin/Release/netcoreapp2.0/PublishesSelfContainedFalse/MSBuildTestApp/bin/Release/netcoreapp2.0/linux-x64/publish/MSBuildTestApp.dll
It was not possible to find any compatible framework version
The specified framework 'Microsoft.NETCore.App', version '2.0.9' was not found.
  - Check application dependencies and target a framework version installed at:
      /
  - Alternatively, install the framework version '2.0.9'.
bash-4.2#
```

**Comment by livarcocc** [comment]
Could you do a binlog fo the test project and share that with us? The test project being under `/build/cli/test/dotnet-publish.Tests/bin/Release/netcoreapp2.0/PublishesSelfContainedFalse/MSBuildTestApp`. Also, make sure to use the just built CLI for that.

Can you also share the contents of the `Microsoft.NETCoreSdk.BundledVersions.props` file under the just built SDK. It should be under `/build/cli/artifacts/linux-x64/stage2/sdk/<version>`. This file contains the implicit framework version we use when building an app.

Basically, I am trying to figure out where the 2.0.9 version is coming from.

**Comment by assyadh** [comment]
Hi,

Here is the build I just ran using the cli previously built:
It is trying to install Microsoft.NETCore.App 2.0.9
```
bash-4.2# ./dotnet build /build/cli/test/dotnet-publish.Tests/bin/Release/netcoreapp2.0/PublishesSelfContainedFalse/MSBuildTestApp/ /bl:out.binlog
Microsoft (R) Build Engine version 15.7.179.6572 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

/build/cli/artifacts/linux-x64/stage2/sdk/2.1.201-preview-007625/MSBuild.dll /bl:out.binlog /clp:Summary /Logger:Microsoft.DotNet.Tools.MSBuild.MSBuildLogger,/build/cli/artifacts/linux-x64/stage2/sdk/2.1.201-preview-007625/dotnet.dll /m /restore /t:Build /v:m /build/cli/test/dotnet-publish.Tests/bin/Release/netcoreapp2.0/PublishesSelfContainedFalse/MSBuildTestApp/MSBuildTestApp.csproj
  Restoring packages for /build/cli/test/dotnet-publish.Tests/bin/Release/netcoreapp2.0/PublishesSelfContainedFalse/MSBuildTestApp/MSBuildTestApp.csproj...
  Restoring packages for /build/cli/test/dotnet-publish.Tests/bin/Release/netcoreapp2.0/PublishesSelfContainedFalse/MSBuildTestApp/MSBuildTestApp.csproj...
  Installing dotnet-portable 1.0.0.
  Restore completed in 129.81 ms for /build/cli/test/dotnet-publish.Tests/bin/Release/netcoreapp2.0/PublishesSelfContainedFalse/MSBuildTestApp/MSBuildTestApp.csproj.
  Restoring packages for /build/cli/test/dotnet-publish.Tests/bin/Release/netcoreapp2.0/PublishesSelfContainedFalse/MSBuildTestApp/MSBuildTestApp.csproj...
  Installing dotnet-prefercliruntime 1.0.0.
  Restore completed in 6.8 ms for /build/cli/test/dotnet-publish.Tests/bin/Release/netcoreapp2.0/PublishesSelfContainedFalse/MSBuildTestApp/MSBuildTestApp.csproj.
  Installing Microsoft.NETCore.DotNetAppHost 2.0.9.
  Installing Microsoft.NETCore.DotNetHostResolver 2.0.9.
  Installing NETStandard.Library 2.0.3.
  Installing Microsoft.NETCore.DotNetHostPolicy 2.0.9.
  Installing Microsoft.NETCore.App 2.0.9.
  Installing runtime.win-x86.Microsoft.NETCore.DotNetAppHost 2.0.9.
  Installing runtime.win-x86.Microsoft.NETCore.DotNetHostResolver 2.0.9.
  Installing runtime.win-x86.Microsoft.NETCore.DotNetHostPolicy 2.0.9.
  Installing runtime.win-x86.Microsoft.NETCore.App 2.0.9.
  Installing runtime.win-x64.Microsoft.NETCore.DotNetAppHost 2.0.9.
  Installing runtime.win-x64.Microsoft.NETCore.DotNetHostResolver 2.0.9.
  Installing runtime.win-x64.Microsoft.NETCore.DotNetHostPolicy 2.0.9.
  Installing runtime.win-x64.Microsoft.NETCore.App 2.0.9.
  Installing runtime.osx-x64.Microsoft.NETCore.DotNetAppHost 2.0.9.
  Installing runtime.osx-x64.Microsoft.NETCore.DotNetHostResolver 2.0.9.
  Installing runtime.osx-x64.Microsoft.NETCore.DotNetHostPolicy 2.0.9.
  Installing runtime.osx-x64.Microsoft.NETCore.App 2.0.9.
  Installing runtime.linux-x64.Microsoft.NETCore.DotNetAppHost 2.0.9.
  Installing runtime.linux-x64.Microsoft.NETCore.DotNetHostResolver 2.0.9.
  Installing runtime.linux-x64.Microsoft.NETCore.DotNetHostPolicy 2.0.9.
  Installing runtime.linux-x64.Microsoft.NETCore.App 2.0.9.
  Generating MSBuild file /build/cli/test/dotnet-publish.Tests/bin/Release/netcoreapp2.0/PublishesSelfContainedFalse/MSBuildTestApp/obj/MSBuildTestApp.csproj.nuget.g.props.
  Generating MSBuild file /build/cli/test/dotnet-publish.Tests/bin/Release/netcoreapp2.0/PublishesSelfContainedFalse/MSBuildTestApp/obj/MSBuildTestApp.csproj.nuget.g.targets.
  Restore completed in 14.56 sec for /build/cli/test/dotnet-publish.Tests/bin/Release/netcoreapp2.0/PublishesSelfContainedFalse/MSBuildTestApp/MSBuildTestApp.csproj.
  MSBuildTestApp -> /build/cli/test/dotnet-publish.Tests/bin/Release/netcoreapp2.0/PublishesSelfContainedFalse/MSBuildTestApp/bin/Debug/netcoreapp2.0/MSBuildTestApp.dll

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:17.03
```

Content of the Microsoft.NETCoreSdk.BundledVersions.props:
```
<Project>
  <PropertyGroup>
    <BundledNETCoreAppTargetFrameworkVersion>2.0</BundledNETCoreAppTargetFrameworkVersion>
    <BundledNETCoreAppPackageVersion>2.0.7</BundledNETCoreAppPackageVersion>
    <BundledNETStandardTargetFrameworkVersion>2.0</BundledNETStandardTargetFrameworkVersion>
    <BundledNETStandardPackageVersion>2.0.1</BundledNETStandardPackageVersion>
  </PropertyGroup>
</Project>
```

Is there a way I could MP you to share the binlog?

Thanks!

**Comment by livarcocc** [comment]
I see why this is happening now. There were some 2.0.9 Microsoft.NetCore.App packages published to myget and this test is rolling forward to it, even though the CLI is not targeting it yet.

I am looking into why this package was published and what else we can do to fix it.

Thanks for reporting the issue.

**Comment by assyadh** [comment]
Thanks for looking into this.

**Comment by livarcocc** [comment]
This should be fixed now.
